utils/fragmented_temporary_buffer.hh

```diff
@@ -193,6 +193,7 @@ public:
     }
 
     bool empty() const noexcept { return !size_bytes(); }
+    size_t size() const noexcept { return _total_size; }
```

Comment:
This is problematic. The view is also an std::ranges::range, adding size() would make it std::ranges::sized_range, but with the wrong result for size(). (we're not using std::ranges at the moment but the same principle applies).

fragmented_temporary_buffer represents two things, a container of fragments and a container of bytes, we should be careful not to mix them.

Reply:
/cc @michoecho

Reply:
The size() was added solely to make compound_type::serialize_value work with fragmented_temporary_buffer::view, we can change serialize_value() to use size_bytes() instead of size() when it's appropriate (currently it works with managed_bytes_view because size() is the same as size_bytes() there)

Reply:
You are referring to compound.hh's serialized_value?

Reply:
If so, I think it's better to indeed change it to use size_bytes (and add a constraint to serialize_value to make it clear what it expects).

Reply:
> You are referring to compound.hh's serialized_value?

yes, it's compound_type::serialize_value()
> If so, I think it's better to indeed change it to use size_bytes (and add a constraint to serialize_value to make it clear what it expects).

Ok